build-tests/sparo-real-repo-test/src/start-test.ts

```diff
@@ -0,0 +1,113 @@
+  const {
+    heftTaskSession: {
+      logger,
+      parameters: { production }
+    },
+    heftConfiguration: { buildFolderPath }
+  } = runScriptOptions;
+
+  const temporaryDirectory: string = path.resolve(buildFolderPath, 'temp');
+  const testRepoURL: string = 'https://github.com/tiktok/sparo.git';
+  const repoFolder: string = path.resolve(temporaryDirectory, 'sparo');
+
+  await FileSystem.deleteFolderAsync(repoFolder);
+
+  const commandDefinitions: ICommandDefinition[] = [
+    // sparo clone git@github.com:tiktok/sparo.git --branch build-test
+    {
+      kind: 'sparo-command',
+      name: 'clone',
+      args: ['clone', testRepoURL, '--branch', 'build-test'],
+      currentWorkingDirectory: temporaryDirectory
+    },
+    // sparo init-profile --profile my-build-test
+    {
+      kind: 'sparo-command',
+      name: 'init-profile',
+      args: ['init-profile', '--profile', 'my-build-test'],
+      currentWorkingDirectory: repoFolder
+    },
+    // sparo checkout --profile my-build-test - extra step to checkout an empty profile
+    {
+      kind: 'sparo-command',
+      name: 'checkout-empty-profile',
+      args: ['checkout', '--profile', 'my-build-test'],
+      currentWorkingDirectory: repoFolder
+    },
+    // Prepare my-build-test profile
+    {
+      kind: 'custom-callback',
+      name: 'prepare-my-build-test-profile',
+      callback: async () => {
+        await FileSystem.writeFileAsync(
+          path.resolve(repoFolder, 'common/sparo-profiles/my-build-test.json'),
+          JSON.stringify(
+            {
+              $schema: 'https://tiktok.github.io/sparo/schemas/sparo-profile.schema.json',
+              selections: [
+                {
+                  selector: '--to',
+                  argument: 'sparo'
+                }
+              ],
+              includeFolders: [],
+              excludeFolders: []
+            },
+            null,
+            2
+          )
+        );
+      }
+    },
+    // sparo checkout --profile my-build-test
+    {
+      kind: 'sparo-command',
+      name: 'checkout-profile',
+      args: ['checkout', '--profile', 'my-build-test'],
+      currentWorkingDirectory: repoFolder
+    },
+    // sparo list-profiles
+    {
+      kind: 'sparo-command',
+      name: 'list-profiles',
+      args: ['list-profiles'],
+      currentWorkingDirectory: repoFolder
+    },
+    // sparo list-profiles --project sparo-lib
+    {
+      kind: 'sparo-command',
+      name: 'list-profiles-with-project',
+      args: ['list-profiles', '--project', 'sparo-lib'],
+      currentWorkingDirectory: repoFolder
+    }
+  ];
+
+  await executeCommandsAndCollectOutputs({
+    buildFolderPath,
+    commandDefinitions
+  });
+
+  await updateOrCompareOutputs({
+    buildFolderPath,
+    logger,
+    production
+  });
+
+  // Clean up the temporary directory in CI builds, but leave it for local debugging
+  if (production) {
+    await FileSystem.deleteFolderAsync(repoFolder);
```

Comment:
(Why is this needed? Won't a CI VM usually get discarded?)

It seems like the best time to delete files is /before/ the operation rather than /after/.

Reply:
Removed